ieeeuoft/hackathon-template
# Review Admin Forms

## Overview

- Adds a proxy model for `registration.Team`, so that it can be registered in the admin site again with different inlines. The migrations don't do anything to the database.
- Adds forms to the admin site for reviewing applications, by overriding the `ModelForm` that would normally be used in `ApplicationInline`.
- Adds a custom formset for handling the application form, just to pass in a request object.


## Unit Tests Created
Form tests:
- Initialization given an existing review, no initialization with no review
- Doesn't allow review changes after a decision has been sent
- Saving a review for the first time
- Overwriting an existing review (the reviewer will be changed)
- Saving based off the `commit` parameter

Admin list view tests:
- Is accessible or inaccessible depending on permissions
- Shows member count and whether the team has been fully reviewed or not
- Filters on team reviewed or not

Admin change view tests:
- Is accessible or inaccessible depending on permissions
- Successfully submits reviews
- Renders existing reviews

## Steps to QA
- Create one or more users, and submit applications for them.
- Have some of the users join a team
- Go to the "Teams" section of the Review app on the admin site.
- Teams should appear in the list view.
    - Number of members should be correct.
    - The team should only be marked as reviewed if all members have been reviewed.
- Open the change page for a particular team.
- Submit a review. Everything should be updated properly.
- Go to the review model in the admin site, and set a decision sent date. Try to submit a new review through the team page (change some fields). You should not be able to.
